### PR TITLE
Expire stale conntrack connections

### PR DESCRIPTION
**Pull request by leeavital**
### What does this PR do?

A percentage of conntrack record close events are not being propogated to the conntracker module which will cause us to hit our 65k record limit, so we need to manually expire them after 15 minutes

This PR adds simple expiry logic to remove conntrack entries if they haven't been read in a 15 minute window.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by leeavital** [comment]
Stats from a system-probe running for 30 minutes in a test environment:

```
{
  "conntrack": {
    "expires": 157,
    "gets_total": 4983,
    "nanoseconds_per_get": 1360,
    "nanoseconds_per_register": 4701,
    "registers_total": 1395,
    "short_term_buffer_size": 0,
    "state_size": 168
  },
  "state": {
    "clients": {
      "1": {
        "closed_connections": 7,
        "last_fetch": 1559852120,
        "stats": 36
      }
    },
    "current_time": 1559852125,
    "latest_bpf_time_ns": 1704767904121579,
    "telemetry": {
      "closed_conn_dropped": 0,
      "closed_conn_polling_lost": 0,
      "closed_conn_polling_received": 79,
      "conn_dropped": 0,
      "expired_tcp_conns": 0,
      "ok_conns_skipped": 0,
      "stats_resets": 0,
      "time_sync_collisions": 0,
      "unordered_conns": 0
    }
  }
}

```

File: pkg/ebpf/netlink/conntracker.go [hunk 1]
```diff
@@ -278,10 +293,16 @@ func (ctr *realConntracker) compact() {
...
 	defer ctr.Unlock()
 
+	gen := getCurrentGeneration(compactInterval, time.Now().UnixNano())
+
 	// https://github.com/golang/go/issues/20135
-	copied := make(map[connKey]*IPTranslation, len(ctr.state))
+	copied := make(map[connKey]*connValue, len(ctr.state))
 	for k, v := range ctr.state {
-		copied[k] = v
+		if v.expGeneration != gen {
```
Comment (p-lambert) [inline pkg/ebpf/netlink/conntracker.go]: shouldn't this be `v.expGeneration <= gen`?

Comment (leeavital) [reply pkg/ebpf/netlink/conntracker.go]: It should not, because they roll over at 255

Comment (p-lambert) [reply pkg/ebpf/netlink/conntracker.go]: oh, right 🤦‍♂

File: pkg/ebpf/netlink/conntracker.go [hunk 2]
```diff
@@ -20,6 +20,8 @@ import (
 
 const (
 	initializationTimeout = time.Second * 10
+
+	compactInterval = time.Minute * 5
```
Comment (sunhay) [inline pkg/ebpf/netlink/conntracker.go]: Chatted in person but I think it's safer to have this <= the expiry interval, so 4 minutes would be good.

Comment (leeavital) [reply pkg/ebpf/netlink/conntracker.go]: Done

Comment (p-lambert) [reply pkg/ebpf/netlink/conntracker.go]: just curious, was the motivation to not "miss" a generation expiration in case the compaction call is delayed?

Comment (leeavital) [reply pkg/ebpf/netlink/conntracker.go]: It would otherwise take ~20 hours to roll around and eventually clean up the record.

Comment (sunhay) [reply pkg/ebpf/netlink/conntracker.go]: That was exactly the case @p-lambert :)

**Review by sunhay — DISMISSED** [review]

**Comment by sunhay** [comment]
Sorry - looks like I added some conflicts with the protobuf revert

**Review by sunhay — APPROVED** [review]

**Pull request merged**